File: flutter-api/app/Models/Transaction.php
<?php

namespace App\Models;

use App\Models\Scopes\AncientScope;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Carbon;

class Transaction extends Model
{
    use HasFactory;

    protected $fillable = [
        'category_id',
        'transaction_date',
        'amount',
        'description',
        'user_id',
    ];

    protected $casts = ['transaction_date' => 'datetime',];

    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    public function setAmountAttribute($value)
    {
        $this->attributes['amount'] = $value * 100;
    }

    public function setTransactionDateAttrib($value)
    {
        $this->attributes['transaction_date'] = Carbon::createFromFormat('m/d/y', $value)->format('Y-m-d');
    }

    protected static function booted(): void
    {
        static::addGlobalScope(new AncientScope);
    }
}

File: flutter_application_1/lib/Models/transaction.dart
class Transaction {
  int id;
  int categoryId;
  String description;
  String amount;
  String transactionDate;
  String createdAt;

  Transaction({
    required this.id,
    required this.categoryId,
    required this.description,
    required this.amount,
    required this.transactionDate,
    required this.createdAt,
  });

  factory Transaction.fromJson(Map<String, dynamic> json) {
    return Transaction(
      id: json['id'] ?? 0,
      categoryId: json['category_id'] ?? 0,
      description: json['description'] ?? '',
      amount: json['amount'] ?? '0.00',
      transactionDate: json['transaction_date'] ?? '',
      createdAt: json['created_at'] ?? '',
    );
  }
}
